Tolerate long /proc/cpuinfo lines

ddn0 opened the pull request:
Some machines can have quite long /proc/cpuinfo lines (e.g., lots of CPU
flags). This previously exceeded the preallocated line size and
prevented successful parsing.

ddn0 commented:
Probably should add something about setting the libcxx option in the conan profile.

yuhc commented:
When having multiple LLVMs, adding `-DLLVM_DIR` will be helpful.

`cmake -DCMAKE_TOOLCHAIN_FILE=conan_paths.cmake -DLLVM_DIR="$(llvm-config-10 --cmakedir)" ..`

And there seems no need to set llvm-10 as default.

l-hoang commented:
re: distributed/hetero Galois README

Yes, I actually updated the distributed README in the main Galois branch a few days ago. It should be merged into this fork soon.

yuhc commented:
NVCC 10.1/10.2 doesn't support gcc > 8. It seems the recommended gcc version for compiling GPU version is 7.
I'm compiling the GPU benchmarks with `cmake -DCMAKE_TOOLCHAIN_FILE=conan_paths.cmake -DLLVM_DIR="$(llvm-config-10 --cmakedir)" -DCMAKE_CUDA_COMPILER="/usr/local/cuda-10.2/bin/nvcc" -ENABLE_GPU ..`

yuhc commented:
I'm not sure what happened, but after I ran `cmake` and `make` in the build directory, sometimes my bash environment (at least $PATH) was reset.

ddn0 commented:
I'll add the cmake/llvm workarounds here. but let's defer GPU instructions to a separate PR once we figure out the best practices should be (i.e., the happy path to recommend to others).

ddn0 committed:
Tolerate long /proc/cpuinfo lines

Some machines can have quite long /proc/cpuinfo lines (e.g., lots of CPU
flags). This previously exceeded the preallocated line size and
prevented successful parsing.

gurbinder533 reviewed (approved):
Beautiful.
